### PR TITLE
144 rocky8

### PR DESCRIPTION
**Pull request by minorOffense**
See #144

### CODE REVIEW

**Comment by djjudas21** [comment]
Thanks for your contribution. This passes tests and looks good to me - I assume it's working in your environment @minorOffense? I'm not currently using FreeRADIUS at the moment so I don't have a test platform.

**Comment by minorOffense** [comment]
The service started with that in place yes on Rocky 8. I didn't test CentOS Stream or RHEL8.

**Comment by djjudas21** [comment]
Thanks @minorOffense 

@nward or @amateo are either of you guys on CentOS/RHEL? I guess they're highly likely to be the same as Rocky 8, but you never know... :smile:

**Comment by nward** [comment]
I have an open question on #144:
> Does FreeRADIUS run in the default configuration without a certificate? Certificates are of course not required in all RADIUS use cases.

@minorOffense Are you in a position to answer this? I am concerned that it could stop FreeRADIUS's default configuration starting if certs are never generated?
I wonder if we should generate certs if none are provided - i.e. rather than relying on bootstrap running and us then purging the certs dir.

**Comment by nward** [comment]
I note that EL9 doesn't run bootstrap:
```
* Wed Mar 10 2021 Robbie Harwood <rharwood@redhat.com> - 3.0.21-11
- Disable automatic bootstrap
```

**Comment by minorOffense** [comment]
If we know the certs that bootstrap generate I’d add a series of file definitions to ensure present on those files. So the service doesn’t break on install. 

That’s what we did in our internal puppet code.  An exec on bootstrap then register the files it generates.

File: manifests/init.pp [hunk 1]
```diff
@@ -109,6 +109,15 @@
     notify  => Service[$freeradius::fr_service],
   }
 
+  # Add systemd unit to override default file on RHEL8 systems.
+  if ($facts['os']['family'] == 'RedHat' and $facts['os']['release']['major'] == '8') {
+    systemd::dropin_file { 'remove_bootstrap.conf':
+      ensure  => present,
+      unit    => 'radiusd.service', #@todo programmatically determine the service name
+      content => template('freeradius/systemd_dropin_rhel8.erb'),
+    }
```
Comment (nward) [inline manifests/init.pp:118]: Let's rename this to be unique and make the filename not RHEL8 specific, as this impacts any EL8 and maybe other distros too, perhaps something like:

```suggestion
    systemd::dropin_file { 'freeradius remove bootstrap':
      filename => remove_bootstrap.conf,
      ensure   => present,
      unit     => 'radiusd.service', #@todo programmatically determine the service name
      content  => template('freeradius/systemd_dropin_remove_bootstrap.erb'),
    }
```

Comment (nward) [reply manifests/init.pp:118]: Also - do you want to turn this todo in to an issue, I don't think we can solve that in this PR

File: templates/systemd_dropin_rhel8.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[Service]
+ExecStartPre=
+ExecStartPre=-/bin/chown -R radiusd.radiusd /var/run/radiusd
+ExecStartPre=/usr/sbin/radiusd -C
```
Comment (nward) [inline templates/systemd_dropin_rhel8.erb:4]: As a style thing, can we add a newline at the end of this file and potentially update the filename

File: metadata.json [hunk 3]
```diff
@@ -8,6 +8,10 @@
   "project_page": "https://github.com/djjudas21/puppet-freeradius",
   "issues_url": "https://github.com/djjudas21/puppet-freeradius/issues",
   "dependencies": [
+    {
+      "name": "puppet/systemd",
+      "version_requirement": ">=3.0.0 <4.0.0"
```
Comment (nward) [inline metadata.json:13]: I think we need to be `>=1.1.1` in order to get `systemd::dropin_file` with the params we're using (`filename` which I've suggested above is introduced in `1.1.1`, `systemd::dropin_file` without it is in `1.0.0`). I don't know if we want to test all the way back there, but I think if we can support older we should, as some users will be on older versions of `puppet/systemd`

Comment (nward) [reply metadata.json:13]: Also note that `camptocamp/systemd` is currently referenced here and in `.fixtures.yml`, so we would need to remove/update those references and ensure this is compatible (from memory, `camptocamp/systemd` became `puppet/systemd`?)

**Review by nward — COMMENTED** [review]
Here's my thoughts, this looks good so far, nice work! I have a few tweaks suggested.

Do you want to have a stab at some tests for this? It would go in `spec/classes/freeradius_spec.rb`, and check out `spec/classes/radsniff_spec.rb` for some tests which run differently with different facts (osfamily in that case).

File: manifests/init.pp (hunk 1 above)
Comment (nward) [reply manifests/init.pp:118]: > @nward or @amateo are either of you guys on CentOS/RHEL? I guess they're highly likely to be the same as Rocky 8, but you never know... 😄

I actually do have some real world licensed RHEL8 boxes, but not running FreeRADIUS :( I am happy with this being tested on Rocky8 - I'm all over those RPMs and they're all the same in this regard!

**Comment by jnk0** [comment]
We would appreciate to see this PR merged. Already tested it on a couple of Rocky Linux 8 boxes.

**Comment by djjudas21** [comment]
Consensus seems to be that this works so it should be safe. @nward, you left some review comments last time you looked at it. Are you satisfied that they've been addressed?

**Comment by nward** [comment]
> Consensus seems to be that this works so it should be safe. @nward, you left some review comments last time you looked at it. Are you satisfied that they've been addressed?

They don't appear to have been addressed.

I have some day job work doing FreeRADIUS 3.2 on EL9 with Puppet over the next 2 weeks, so I'll tidy it up and do any further EL9 and FR3.2 changes, if there's no objection.

**Comment by nward** [comment]
I've made changes to resolve these issues here: https://github.com/djjudas21/puppet-freeradius/tree/144-rocky8

It looks like @minorOffense didn't allow maintainers to push to their branch - so I think the best approach is to close this, and create a new PR?

**Comment by djjudas21** [comment]
I've created a new PR #194 which contains all of these commits plus @nward's fixes. Closing this in favour of #194

**Pull request closed without merging**